src/astra_driver.cpp

```diff
@@ -388,8 +389,53 @@ void AstraDriver::applyConfigToOpenNIDevice()
     ROS_ERROR("Could not set auto white balance. Reason: %s", exception.what());
   }
 
+  // Workaound for https://github.com/ros-drivers/openni2_camera/issues/51
+  // This is only needed when any of the 3 setting change.  For simplicity
+  // this check is always performed and exposure set.
+  if ((!auto_exposure_ && !auto_white_balance_) && exposure_ != 0)
```

Comment:
Did you verify this same "work-around" is required for the astra camera?

Reply:
Yes, without this workaround it will not set the exposure correctly on start or if either auto_exposure or auto_white_balance are disabled.

Reply:
This makes me sad.